sklearn/decomposition/tests/test_pca.py

```diff
@@ -260,11 +261,12 @@ def test_singular_values():
               random_state=rng).fit(X)
     apca = PCA(n_components=2, svd_solver='arpack',
                random_state=rng).fit(X)
-    rpca = PCA(n_components=2, svd_solver='randomized',
+    # Increase the number of power iterations to get greater accuracy in tests
+    rpca = PCA(n_components=2, svd_solver='randomized', iterated_power=40,
                random_state=rng).fit(X)
-    assert_array_almost_equal(pca.singular_values_, apca.singular_values_, 12)
-    assert_array_almost_equal(pca.singular_values_, rpca.singular_values_, 1)
-    assert_array_almost_equal(apca.singular_values_, rpca.singular_values_, 1)
+    assert_allclose(pca.singular_values_, apca.singular_values_, atol=12)
```

Comment:
This should not be `atol=12`. `assert_allclose(0, 12, atol=12)` passes.

If `assert_array_almost_equal(pca.singular_values_, apca.singular_values_, 12)` used to pass, then you should not touch `atol`, and set `rtol` to some very small value (smallest value higher or equal to the default that makes the test pass)

Reply:
Thanks for catching this.